### PR TITLE
Remove unused GPS fail circuit breaker

### PR DESCRIPTION
**Pull request by CarlOlsson**
Noticed that CBRK_GPSFAIL was not used anymore.

I did not find why the GPS loss failsafe was removed?

### CODE REVIEW

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -2331,8 +2331,6 @@ Commander::get_circuit_breaker_params()
 			CBRK_AIRSPD_CHK_KEY);
 	status_flags.circuit_breaker_engaged_enginefailure_check = circuit_breaker_enabled_by_val(_param_cbrk_enginefail.get(),
 			CBRK_ENGINEFAIL_KEY);
-	status_flags.circuit_breaker_engaged_gpsfailure_check = circuit_breaker_enabled_by_val(_param_cbrk_gpsfail.get(),
```
Comment (dagar) [inline src/modules/commander/Commander.cpp]: Remove `circuit_breaker_engaged_gpsfailure_check` from vehicle_status_flags?

**Comment by Antiheavy** [comment]
> I did not find why the GPS loss failsafe was removed?

I'll be interested to learn if you find out.

**Comment by hamishwillee** [comment]
Me too. If this goes in there are corresponding docs fixes required: http://docs.px4.io/master/en/config/safety.html#position-gps-loss-failsafe

**Comment by CarlOlsson** [comment]
> I'll be interested to learn if you find out.

Maybe it was a mistake in the commander rework. The failsafe we use on our VTOL is the following:
If all GPS modules lose fix for more than a second, trigger RTH and dead reckon towards home. If uncertainty in global position grows too large, trigger transition to hover and descend.

**Comment by Antiheavy** [comment]
I'm still confused about the original intent of this PR.  I believe there are valid PX4 use cases that do not use GPS at all.  Wouldn't they need this?

**Comment by CarlOlsson** [comment]
> I'm still confused about the original intent of this PR. I believe there are valid PX4 use cases that do not use GPS at all. Wouldn't they need this?

The PR just removes variables that are not used in the code. My question in the first post was simply, why they are not used anymore

**Comment by dagar** [comment]
The failsafe applies to the global position estimate (what the controllers use). 

I think one aspect of the system we're missing is a notion of optional/required components. For everything non-optional (eg GPS, multiple IMUs, etc) we would block arming if they're missing.

Additionally we should probably add something simple to catch an actual loss of GPS (if it's required), then notify the operator and trigger an appropriate action before the global position estimate becomes unusable.

At one point we had loss of GPS directly trigger the `AUTO_LANDGPSFAIL` navigation state, but that's isn't correct.

**Comment by Antiheavy** [comment]
I could be wrong but didn't the circuit breaker do two things: 1) remove GPS from the pre-flight/pre-arm checks, and 2) bypass the GPS Failure Navigation (a.k.a. global position failsafe) ?



> I think one aspect of the system we're missing is a notion of optional/required components. For everything non-optional (eg GPS, multiple IMUs, etc) we would block arming if they're missing.


I use the term "external sensors" to describe things like GPS, Mag, Diff Press, etc for sensors that are not physically integrated with the main flight controller.  IMHO the way PX4 treats these needs improving and made more consistent.  We have some ideas here and will make a separate discussion post.



> Additionally we should probably add something simple to catch an actual loss of GPS (if it's required), then notify the operator and trigger an appropriate action before the global position estimate becomes unusable.

Yes!  We actually added some custom code that does exactly this.  For example, in PX4 Master you can lose GPS lock on the ground before takeoff and it will still let you arm and takeoff, even if you have COM_ARM_WO_GPS enabled (because it's based on EKF's global position estimate, not actual GPS lock - this is a "bug" IMHO).  We are trying to decide how to bring it upstream as we need to rebase on newer Master first.

**Comment by Antiheavy** [comment]
> Yes! We actually added some custom code that does exactly this.

FYI @Kjkinney

**Comment by hamishwillee** [comment]
Did any action come out of this discussion ^^^. I'm just removing `CBRK_GPSFAIL` from the docs now. Thankfully the link checker told me about this ... eventually

**Pull request merged**